jetty-util/src/main/java/org/eclipse/jetty/util/resource/PathResource.java

```diff
@@ -564,6 +568,43 @@ public void copyTo(File destination) throws IOException
         }
     }
 
+    /**
+     * @param outputStream the output stream to write to
+     * @param start First byte to write
+     * @param count Bytes to write or -1 for all of them.
+     * @throws IOException if unable to copy the Resource to the output
+     */
+    @Override
+    public void writeTo(OutputStream outputStream, long start, long count)
+        throws IOException
+    {
+        long length = count;
+
+        if (count < 0)
+        {
+            length = Files.size(path) - start;
+        }
+
+        try (SeekableByteChannel channel = Files.newByteChannel(path, StandardOpenOption.READ))
+        {
+            ByteBuffer buffer = BufferUtil.allocate(IO.bufferSize);
+            channel.position(start);
+
+            // copy from channel to output stream
+            long readTotal = 0;
+            while (readTotal < length)
+            {
+                BufferUtil.flipToFill(buffer);
+                int size = (int)Math.min(IO.bufferSize, length - readTotal);
+                buffer.limit(size);
+                int readLen = channel.read(buffer);
+                BufferUtil.flipToFlush(buffer, 0);
```

Comment:
yay you're using `BufferUtil`.... but a slight niggle... it should either be:
```java
   int p = BufferUtil.flipToFill(buffer);
   ...
   BufferUtil.flipToFlush(buffer, p);
```
if you might have an incomplete flush OR
```java
   BufferUtil.clearToFill(buffer);
   ...
   BufferUtil.flipToFlush(buffer, 0);
```
if you know you will always flush completely (as is this case)

Reply:
Done, fix in commit 7a867c0ff02970dbc9b626eb76c1306f5f622f17